slowapi/extension.py

```diff
@@ -646,7 +654,11 @@ def sync_wrapper(*args: Any, **kwargs: Any) -> Response:
                         self._check_request_limit(request, func, False)
                         request.state._rate_limiting_complete = True
                     response = func(*args, **kwargs)
-                    self._inject_headers(response, request.state.view_rate_limit)
+                    if not isinstance(response, Response):
```

Comment:
I'm not sure why the test here does not check for `self._headers_enabled` like above. If there's a reason, it's probably worth mentioning it here (and it probably means my code was not consistent :disappointed: ), but if not, I'd argue the test should be the same for both the sync and async versions. If so, I'd be in favour of having the extra check on both sides, as it saves a method call if headers are not enabled.

Reply:
I removed the test in the previous actually, `_inject_headers` already wraps the check if the headers should be written or not.